### PR TITLE
Fix Local SDK nil Players with test

### PR DESCRIPTION
**Pull request by aLekSer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature

/kind hotfix

**What this PR does / Why we need it**:
There was a situation when additional call in testMode leads to
SegFault, a call when gs.Status.Players is nil. Added tests for different situations with and without files.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like you still have a data race:

```
==================
WARNING: DATA RACE
Read at 0x000002caea18 by goroutine 25:
  agones.dev/agones/pkg/sdkserver.NewLocalSDKServer()
      /go/src/agones.dev/agones/pkg/sdkserver/localsdk.go:137 +0x49f
  agones.dev/agones/pkg/sdkserver.TestSDKConformanceFunctionality()
      /go/src/agones.dev/agones/pkg/sdkserver/localsdk_test.go:551 +0x5f
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199

Previous write at 0x000002caea18 by goroutine 26:
  agones.dev/agones/pkg/sdkserver.NewLocalSDKServer()
      /go/src/agones.dev/agones/pkg/sdkserver/localsdk.go:138 +0x3ed
  agones.dev/agones/pkg/sdkserver.TestAlphaSDKConformanceFunctionality()
      /go/src/agones.dev/agones/pkg/sdkserver/localsdk_test.go:585 +0x5c
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199

Goroutine 25 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1117 +0x2ff
  main.main()
      _testmain.go:96 +0x223

Goroutine 26 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1117 +0x2ff
  main.main()
      _testmain.go:96 +0x223
==================
```

**Comment by markmandel** [comment]
Is this a blocker for 1.6.0?

**Comment by aLekSer** [comment]
Let me fix this data race. I think this could be fixed before the release, someone could be using SDK conformance test local to verify this new feature.

**Comment by markmandel** [comment]
Sound good. I'll take a quick review now too so we can get this through quick. :+1:

File: build/includes/sdk.mk [hunk 1]
```diff
@@ -126,9 +126,10 @@ ensure-build-sdk-image:
 # SDK client test against it. Useful for test development
 run-sdk-conformance-local: TIMEOUT ?= 30
 run-sdk-conformance-local: TESTS ?= ready,allocate,setlabel,setannotation,gameserver,health,shutdown,watch,reserve
+run-sdk-conformance-local: FEATURE_GATES ?=
 run-sdk-conformance-local: ensure-agones-sdk-image
 	docker run -e "ADDRESS=" -p 9357:9357 -p 9358:9358 \
-	 -e "TEST=$(TESTS)" -e "TIMEOUT=$(TIMEOUT)" $(sidecar_tag)
+	 -e "TEST=$(TESTS)" -e "TIMEOUT=$(TIMEOUT)" -e "FEATURE_GATES=$(FEATURE_GATES)" $(sidecar_tag)
```
Comment (markmandel) [inline build/includes/sdk.mk:132]: oooh! I didn't even realise we had a local conformance test. Nice catch!

Do we want to do what we do in the other conformance tests with the new functionality -- i.e. have one test run with the feature gates all enabled, and one without?

File: pkg/sdkserver/localsdk.go [hunk 2]
```diff
@@ -197,8 +200,14 @@ func (l *LocalSDKServer) recordRequestWithValue(request string, value string, ob
 		case "UID":
 			fieldVal = l.gs.ObjectMeta.Uid
 		case "PlayerCapacity":
+			if !runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
```
Comment (markmandel) [inline pkg/sdkserver/localsdk.go:203]: Excellent catch. Nice!

File: pkg/sdkserver/localsdk_test.go [hunk 3]
```diff
@@ -338,115 +339,170 @@ func TestLocalSDKServerPlayerConnectAndDisconnect(t *testing.T) {
...
-
-	count, err = l.GetPlayerCount(context.Background(), e)
-	assert.NoError(t, err)
-	assert.Equal(t, int64(1), count.Count)
-
-	expected := &sdk.GameServer_Status_PlayerStatus{
-		Count:    1,
-		Capacity: 1,
-		Ids:      []string{id.PlayerID},
+	// nolint: maligned
```
Comment (markmandel) [inline pkg/sdkserver/localsdk_test.go:343]: Pretty! Love it.

File: build/includes/sdk.mk (hunk 1 above)
Comment (aLekSer) [reply build/includes/sdk.mk:132]: Yes, that might be reasonable. I have added unit tests with and w/o FeatureGates at least for this new feature.

File: pkg/sdkserver/localsdk.go [hunk 4]
```diff
@@ -89,9 +91,10 @@ type LocalSDKServer struct {
 
 // NewLocalSDKServer returns the default LocalSDKServer
 func NewLocalSDKServer(filePath string) (*LocalSDKServer, error) {
+	gs := defaultGs()
 	l := &LocalSDKServer{
 		gsMutex:         sync.RWMutex{},
-		gs:              defaultGs,
+		gs:              gs,
```
Comment (markmandel) [inline pkg/sdkserver/localsdk.go:97]:
```suggestion
		gs:              defaultGs(),
```

Having it inline seems nicer?

**Pull request merged**